pentaho/src/main/java/org/pentaho/cdf/CdfApi.java

```diff
@@ -372,10 +373,13 @@ protected String getConfiguration( String path,
     return ContextEngine.getInstance().getConfig( path, parameterMap, inactiveInterval );
   }
 
-  protected String buildFullServerUrl( String protocol, String serverName, int serverPort  ) {
-    String p = "http";
+  protected String buildFullServerUrl( String protocol, String serverName, int serverPort, String scheme ) {
```

Comment:
If your going with an extra parameter, I think using [isSecure](http://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#isSecure()) is the best option because it also works if a load balancer is forwarding the requests using HTTP and has more support across different Java HTTP server implementations. Please see this discussion in [stackoverflow](http://stackoverflow.com/questions/8200853/how-can-i-know-if-the-request-to-the-servlet-was-executed-using-http-or-https).